cmd/onboard.go

```diff
@@ -153,69 +154,108 @@ func doOnboard() error {
 		AllowCredentialReuse: true,
 	})
 	if err != nil {
+		if errors.Is(err, context.Canceled) {
+			return fmt.Errorf("onboarding canceled by user: %w", err)
+		}
 		return err
 	}
 	if newDC == nil {
-		fmt.Println("Credential not updated (Credential Reuse Protocol")
+		slog.Info("Credential not updated")
 		return nil
 	}
 
 	// Store new credential
-	fmt.Println("FIDO Device Onboard Complete")
+	slog.Info("FIDO Device Onboard Complete")
 	return updateCred(*newDC, FDO_STATE_IDLE)
 }
 
-func transferOwnership(ctx context.Context, rvInfo [][]protocol.RvInstruction, conf fdo.TO2Config) (*fdo.DeviceCredential, error) { //nolint:gocyclo
-	var to2URLs []string
-	directives := protocol.ParseDeviceRvInfo(rvInfo)
-	for _, directive := range directives {
-		if !directive.Bypass {
-			continue
+// addJitter adds ±25% randomization to a delay duration as per FDO spec v1.1 section 3.7.
+func addJitter(delay time.Duration) time.Duration {
+	jitterPercent := 0.25 * (2*rand.Float64() - 1) // Random from -0.25 to +0.25 (±25%)
+	jitter := float64(delay) * jitterPercent
+	return delay + time.Duration(jitter)
+}
+
+// applyDelay waits for the specified duration with context cancellation support.
+// Defensive: if ctx is nil (shouldn't happen, but protects against refactoring errors),
+// falls back to uncancellable sleep to avoid panics in IoT devices.
```

Comment:
This seems unnecessarily defensive - allowing a nil context is extremely rare across Go code - and the reasoning is unclear. Why would an IoT device not have a Go context? I believe that both TinyGo and Tamago support it just fine.